conftest.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import os
+from typing import Any
+
+import pytest
+import requests
+
+import openeo
+from openeo.capabilities import ComparableVersion
+
+_log = logging.getLogger(__name__)
+
+os.environ["ENDPOINT"] = "https://openeo.dataspace.copernicus.eu/"
+
+def get_openeo_base_url(version: str = "1.1.0"):
+    try:
+        endpoint = os.environ["ENDPOINT"].rstrip("/")
+    except Exception:
+        raise RuntimeError("Environment variable 'ENDPOINT' should be set"
+                           " with URL pointing to OpenEO backend to test against"
+                           " (e.g. 'http://localhost:8080/' or 'http://openeo-dev.vgt.vito.be/')")
+    return "{e}/openeo/{v}".format(e=endpoint.rstrip("/"), v=version)
+
+
+@pytest.fixture(params=[
+    "1.1.0",
+])
+def api_version(request) -> ComparableVersion:
```

Comment:
I think this api version aspect can be eliminated from this test suite for now.
I would just handle this trough the openeo backend  URL

Reply:
The auth_connection(connection, capfd) fixture relies on the api_version fixture. 

Uncertain how I can eliminate it. Are there any examples of test suites without the api version?

Reply:
I think I managed to fix it

Reply:
I think this conftest.py received too much unnecessary bits from copy-pasting from elsewhere.

Essentially you just need something like this:

```python
import os
import openeo
import pytest


@pytest.fixture
def connection(capfd) -> openeo.Connection:
    """
    Fixture to create an authenticate the connection to the backend under test.

    Authentication is based on `authenticate_oidc`, which works by default for local development (e.g. device flow or refresh tokens)
    but also supports (since python client version 0.18.0) client credentials auth
    if the appropriate env vars are set (OPENEO_AUTH_METHOD, OPENEO_AUTH_CLIENT_ID, OPENEO_AUTH_CLIENT_SECRET,
    and OPENEO_AUTH_PROVIDER_ID)
    """
    openeo_backend_url = os.environ.get(
        "OPENEO_BACKEND_URL",
        "https://openeo.dataspace.copernicus.eu/"
    )

    connection = openeo.connect(openeo_backend_url)

    # Temporarily disable output capturing, to make sure that the OIDC device code instructions are shown.
    with capfd.disabled():
        # Use a shorter max poll time by default
        # to alleviate the default impression that the test seem to hang
        # because of the OIDC device code poll loop.
        max_poll_time = int(os.environ.get("OPENEO_OIDC_DEVICE_CODE_MAX_POLL_TIME") or 30)
        connection.authenticate_oidc(max_poll_time=max_poll_time)
    return connection
```


All the rest is overkill at the moment


Note that I just define a `connection` fixture here, which replaces the original `auth_connection`. For the benchmarks, you always need an authenticated connection, so no need to define both `connection` + `auth_connection`